test/linalg/lu.jl

```diff
@@ -83,10 +83,29 @@ debug && println("(Automatic) Square LU decomposition. eltya: $eltya, eltyb: $el
                 @test norm(a*(lua\c) - c, 1) < ε*κ*n # c is a vector
                 @test norm(a'*(lua'\c) - c, 1) < ε*κ*n # c is a vector
                 @test AbstractArray(lua) ≈ a
-                if eltya <: Real && eltyb <: Real
-                    @test norm(a.'*(lua.'\b) - b,1) < ε*κ*n*2 # Two because the right hand side has two columns
-                    @test norm(a.'*(lua.'\c) - c,1) < ε*κ*n
-                end
+                @test norm(a.'*(lua.'\b) - b,1) < ε*κ*n*2 # Two because the right hand side has two columns
+                @test norm(a.'*(lua.'\c) - c,1) < ε*κ*n
+
+                # Test whether Ax_ldiv_B!(y, LU, x) indeed overwrites y
+                resultT = typeof(oneunit(eltyb) / oneunit(eltya))
+
+                b_dest = similar(b, resultT)
+                c_dest = similar(c, resultT)
+
+                A_ldiv_B!(b_dest, lua, b)
+                A_ldiv_B!(c_dest, lua, c)
+                @test norm(b_dest - lua \ b, 1) < ε*κ*2n
+                @test norm(c_dest - lua \ c, 1) < ε*κ*n
+
+                At_ldiv_B!(b_dest, lua, b)
+                At_ldiv_B!(c_dest, lua, c)
+                @test norm(b_dest - lua.' \ b, 1) < ε*κ*2n
+                @test norm(c_dest - lua.' \ c, 1) < ε*κ*n
+
+                Ac_ldiv_B!(b_dest, lua, b)
+                Ac_ldiv_B!(c_dest, lua, c)
+                @test norm(b_dest - lua' \ b, 1) < ε*κ*2n
+                @test norm(c_dest - lua' \ c, 1) < ε*κ*n
```

Comment:
It is because `A.' \ B` becomes `At_ldiv_B(A, B)` (not `transpose(A) \ B`). Perhaps this will do?
```julia
for (Aq_ldiv_B!, f) in (
        (A_ldiv_B!, \),
        (At_ldiv_B!, At_ldiv_B),
        (Ac_ldiv_B!, Ac_ldiv_B) )
    Aq_ldiv_B!(b_dest, lua, b)
    Aq_ldiv_B!(c_dest, lua, c)
    @test norm(b_dest - f(lua, b), 1) < ε*κ*2n
    @test norm(c_dest - f(lua, c), 1) < ε*κ*n
end
```